docs/reference/landmark-based-facial-expression-recognition.md

```diff
@@ -0,0 +1,450 @@
+  Specifies which checkpoint should be loaded. If it is set to 0, no checkpoints will be loaded.
+- **temp_path**: *str, default=''*  
+  Specifies a path where the algorithm saves the checkpoints and onnx optimized model (if needed).
+- **device**: *{'cpu', 'cuda'}, default='cuda'*  
+  Specifies the device to be used.
+- **num_workers**: *int, default=32*  
+  Specifies the number of workers to be used by the data loader.
+- **epochs**: *int, default=400*  
+  Specifies the number of epochs the training should run for.
+- **experiment_name**: *str, default='pstbln_casia'*  
+  String name to attach to checkpoints.
+- **device_ind**: *list, default=[0]*  
+  List of GPU indices to be used if the device is 'cuda'. 
+- **val_batch_size**: *int, default=256*  
+  Specifies number of samples to be bundled up in a batch during evaluation. This heavily affects memory usage, adjust according to your system.
+- **drop_after_epoch**: *list, default=[1000]*  
+  List of epoch numbers in which the optimizer drops the learning rate.  
+- **start_epoch**: *int, default=0*  
+  Specifies the starting epoch number for training.  
+- **dataset_name**: *str {'AFEW', 'CASIA', 'CK+'}, default='CASIA'*  
+  Specifies the name of dataset that is used for training and evaluation. 
+- **block_size**: *int, default=5*  
+  Specifies the number of output channels (or neurons) that are added to each layer of the network at each progression iteration. 
+- **numblocks**: *int, default=100*  
+  Specifies the maximum number of blocks that are added to each layer of the network at each progression iteration. 
+- **numlayers**: *int, default=10*  
+  Specifies the maximum number of layers that are built for the network.
+- **topology**: *list, default=[]*  
+  Specifies the initial topology of the network. The default is set to [], since the method gets an empty network as input and builds it progressively. 
+- **layer_threshold**: *float, default=1e-4*  
+  Specifies the threshold which is used by the method to identify when it should stop adding new layers.  
+- **block_threshold**: *float, default=1e-4*  
+  Specifies the threshold which is used by the model to identify when it should stop adding new blocks in each layer.  
+  
+
+#### `ProgressiveSpatioTemporalBLNLearner.fit`
+```python
+ProgressiveSpatioTemporalBLNLearner.fit(self, dataset, val_dataset, logging_path, silent, verbose,
+                                        momentum, nesterov, weight_decay, monte_carlo_dropout, 
+                                        mcdo_repeats, train_data_filename,
+                                        train_labels_filename, val_data_filename,
+                                        val_labels_filename)
+```
+
+This method is used for training the algorithm on a train dataset and validating on a val dataset.
+Parameters:
+- **dataset**: *object*  
+Object that holds the training dataset.
+Can be of type `ExternalDataset` or a custom dataset inheriting from `DatasetIterator`.
```

Comment:
It is rendered correctly, but all the lines in the fit method should be indented by 2 spaces.